aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1221,6 +1235,31 @@ static void addmm_impl_cpu_(
   }
 }
 
+Tensor addmm_zerotensor(const Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha){
+  check_addmm(self, mat1, mat2);
+  const auto self_sizes = self.sizes();
+  Tensor result;
+  // beta*self + alpha*(mat1@mat2) = 0
+  if ((mat1._is_zerotensor() || mat2._is_zerotensor() || alpha.equal(0)) &&
+      (self._is_zerotensor() || beta.equal(0))) {
+    result = at::_efficientzerotensor(self_sizes, self.options());
+    return result;
+  }
+  // beta*self
+  else if (mat1._is_zerotensor() || mat2._is_zerotensor() || alpha.equal(0)){
+    result.resize_as_(self);
+    result.copy_(self);
+    return beta.equal(1) ? result : result.mul_(beta);
+  }
+  // alpha*(mat1@mat2)
+  else if (self._is_zerotensor() || beta.equal(0)){
+    result = at::mm(mat1, mat2);
+    return alpha.equal(1) ? result : result.mul_(alpha);
+  } else /* never reached */{
```

Comment:
else is not needed

Reply:
Without the `else` here the compiler will error out as a non-void function without a return.

This seemed more clear than leaving the `else` and keeping the `return`.

Reply:
What about making the last else if an else and leaving the condition as a comment inside of it?

Reply:
Yup that'd work.

Reply:
I second @Lezcano